App/dojo.py

```diff
@@ -369,8 +386,149 @@ def load_people(self, filename):
                     click.secho('Success',fg='green')
         else:
             print('please provide a file')
+                 
+    def save_state(self,db_name='dojo'):
+        '''
+        Persists all the data stored in the app to a 
+        SQLite database. Specifying the --db parameter
+        explicitly stores the data in the sqlite_database 
+        specified. 
+        '''
+        # create engine
+        engine = create_db(db_name)
+        # connect to db
+        Base.metadata.bind = engine
+        Session = sessionmaker()
+        session = Session()
+
+        # function to select all rooms in the Rooms table
+        items = select([Rooms])
+        res = session.execute(items)
+        db_room_list =[item.room.name for item in res]
+
+        for room in self.all_rooms:
+            if room.name not in db_room_list:
+                new_room = Rooms(name=room.name,
+                                type_room = room.room_type,room_capacity = room.capacity)
+                session.add(new_room)
+                session.commit()
+
+        people = select([Persons])
+        response = session.execute(people)
+        dbpersons_list = [item.person.name for item in response]
```

Comment:
another way to check for duplicate if some data is already in the table would be to use a filter statement.